include/nbl/ext/ImGui/ImGui.h

```diff
@@ -21,10 +21,10 @@ class UI final : public core::IReferenceCounted
 	private:
 		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> createDescriptorSetLayout();
 		void createPipeline(video::IGPURenderpass* renderpass, video::IGPUPipelineCache* pipelineCache);
-		void createFontTexture(video::IGPUCommandBuffer* cmdBuffer, video::IQueue* queue);
+		void createFontAtlas2DArrayTexture(video::IGPUCommandBuffer* cmdBuffer, video::IQueue* queue);
```

Comment:
why array?

Reply:
ah ok atlas bakes all fonts into one single big texture on `io.Fonts->GetTexDataAsRGBA32` call, wanted to prepare for indexing which is also possible but we will stay with the atlas though it has some [disadvantages](https://github.com/ocornut/imgui/blob/master/docs/FONTS.md#4-font-atlas-texture-fails-to-upload-to-gpu)

well I need to update the atlas texture's gpu content if new fonts are added/removed before calling new imgui frame request

Reply:
> wanted to prepare for indexing which is also possible but we will stay with the atlas

ok for sure we stay with the atlas thing and no indexing/array for **fonts**, but descriptor indexing for [user defined textures](https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples)

Reply:
> ok for sure we stay with the atlas thing and no indexing/array for fonts

**no** I take that back, `ImFontAtlas::SetTexID()` sets the id for the atlas texture so we will put all into `2DTexture[6969]` and use descriptor indexing, the index can be obtained with `ImDrawCmd`